sorrydb/database/process_sorries.py

```diff
@@ -29,6 +31,40 @@ def should_process_file(lean_file: Path) -> bool:
     return any(term in text for term in ["sorry"])
 
 
+def get_potential_sorry_files(
+    repo_path: Path,
+    is_mathlib: bool = False,
+) -> list[Path]:
+    """Get list of Lean files to process.
+
+    Args:
+        repo_path: Path to the repository root
+        is_mathlib: If True, only include files that differ from master branch
+                    (used for mathlib repository)
+
+    Returns:
+        List of relative paths for each Lean file to process
+    """
+    lean_files = list(repo_path.rglob("*.lean"))
+
+    if is_mathlib:
+        # For mathlib, we only want files that differ from both:
+        # 1. The merge-base (most recent common ancestor with master)
+        # 2. The current master HEAD
+        # Any file on master is guaranteed not to contain sorries.
+        merge_base = get_merge_base(repo_path, "origin/master")
+        diff_base = set(get_changed_files(repo_path, merge_base))
+        diff_head = set(get_changed_files(repo_path, "origin/master"))
+        changed = diff_base.intersection(diff_head)
```

Comment:
Can we leverage the [triple dot diff](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-codegitdiffltoptionsgtltcommitgtltcommitgt--ltpathgtcode-1) here:

```suggestion
        repo = Repo(repo_path)
        changed = repo.git.diff("master...HEAD", name_only=True).splitlines()

```

Reply:
If I understand things correctly this is not logically equivalent to the old code? The old code tested if a file is different from *both* the merge base and the most recent master revision. The latter could be helpful if an earlier revision of the branch had been merged in to master. I do not know if this happens frequently, but I can certainly  imagine such scenarios.

Reply:
Ah, yes. We would be processing unnecessary files in that case, if we didn't compare the most recent master revision as well. I can imagine this happening when someone breaks up a large PR. I agree that we should be more conservative.

Reading this again, your code is also more explicit.